Update SignalFX Handler name + add filtering metrics

prasincs opened the pull request:
This updates the SignalFX Handler to be able to provide a config
`filter_metrics_regex` that's in the form of comma separated list of
`collector_name:metrics_regex` that can be provided in the config
file.

if this regex isn't set, all metrics are sent, otherwise the metrics
will be filtered based on the list.

shortdudey123 commented:
Can you update the doc file for the handler as well since you are changing the doc string? Also, please fix the pep8 issues

prasincs commented:
I have updated the docs and fixed the pep8 issues but since the original didn't have any tests, I haven't added any. If there's a particular format you want the tests in, please provide a link to a sample that can be adapted for the collector.

shortdudey123 commented on docs/handlers/SignalfxHandler.md:
```diff
@@ -14,18 +14,21 @@ Send metrics to signalfx
+get_default_config_help |  | get_default_config_help |
 server_error_interval | 120 | How frequently to send repeated server errors | int
 url | https://ingest.signalfx.com/v2/datapoint | Where to send metrics | str
+filter_metrics_regex | | Comma Separated collector:regex filters| str
```
alphabetize

prasincs replied:
you mean just order them in the table alphabetically?

shortdudey123 replied:
yup

shortdudey123 commented:
Looks fine other than the 1 comment

shortdudey123 commented on src/diamond/handler/signalfx.py:
```diff
@@ -18,6 +18,8 @@
      posting
  * filter_metrics_regex = [optional] comma separated list of collector:regex
      to limit metrics sent to signalfx,  default is to send everything
+ * url = [optional| https://ingest.signalfx.com/v2/datapoint] where to send
```
This needs to be added to the doc too if you have this
